cf_units/__init__.py

```diff
@@ -384,7 +384,22 @@ def suppress_errors():
                 'System', 'udunits2_xml_path',
                 default=os.path.join(sys.prefix, 'share', 'udunits',
                                      'udunits2.xml'))
-            _ud_system = _ut_read_xml(_alt_xml_path.encode())
+            if os.name == 'nt':
+                # On Windows systems we need to read the xml while on the
+                # system drive but we don't want to change the
```

Comment:
Well thanks for considering anyway.  I agree this solution is a bit hacky,
and will look forward to it getting fixed in udunits someday.


Colin Talbert
GIS Analyst and Developer
US Geological Survey

talbertc@usgs.gov

USGS Fort Collins Science Center
2150 Centre Ave. Bldg. C
Fort Collins, CO 80526
(970) 226-9425

USGS North Central Climate Science Center
(970) 492-4283

Work schedule:
Monday       - 7:00 - 3:00  (FORT)
Tuesday       -  7:00 - 3:00  (NC CSC)
Wednesday  -  7:00 - 3:00 (FORT)
Thursday      -  7:00 - 5:00 (FORT)
Friday          -  7:00 - 5:00 (FORT)

On Wed, Aug 3, 2016 at 2:17 AM, Peter Killick <notifications@github.com>
wrote:

> In cf_units/__init__.py
> <https://github.com/SciTools/cf_units/pull/65#discussion_r73297602>:
>
> > @@ -384,7 +384,22 @@ def suppress_errors():
> >                  'System', 'udunits2_xml_path',
> >                  default=os.path.join(sys.prefix, 'share', 'udunits',
> >                                       'udunits2.xml'))
> > -            _ud_system = _ut_read_xml(_alt_xml_path.encode())
> > +            if os.name == 'nt':
> > +                # On Windows systems we need to read the xml while on the
> > +                # system drive but we don't want to change the
>
> This assertion is actually incorrect: you need to read the XML while your
> CWD is *on the drive that the XML is located upon*.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SciTools/cf_units/pull/65/files/e10dd51af251640bab212a0eb690948e857a921a#r73297602>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACJURPA9s8qyhp8mdevA_xJ67aQMF_6Rks5qcE6WgaJpZM4JXW2F>
> .
>